Bio/SeqFeature.py

```diff
@@ -1188,6 +1203,24 @@ def __iter__(self):
             for pos in loc:
                 yield pos
 
+    def __eq__(self, other):
+        """Check if all parts of CompoundLocation are equal to all parts of other CompoundLocation."""
+        if not isinstance(other, CompoundLocation):
+            return False
+        if len(self.parts) != len(other.parts):
+            return False
+        if self.operator != other.operator:
+            return False
+        identical = True
+        for self_part, other_part in zip(self.parts, other.parts):
+            identical = identical and self_part == other_part
+        return identical
```

Comment:
Do you find this form with the local variable ``identical`` clearer than my suggestion, which would short-cut comparing all the parts as soon as one difference is found?

```
for self_part, other_part in zip(self.parts, other.parts):
    if self_part != other_part:
        return False
return True
```